forge-game/src/main/java/forge/game/trigger/TriggerWaiting.java

```diff
@@ -13,6 +16,7 @@ public class TriggerWaiting {
     private TriggerType mode;
     private Map<AbilityKey, Object> params;
     private List<Trigger> triggers = null;
+    private Map<Trigger, Player> controllers = null;
```

Comment:
i don't know if we really need two extra lists, or if they can be combined into one

i mean the trigger list can also be just returned by map.keySet ?

Reply:
Done
I think since TriggerHandler supports duplicates (maybe some trigger with LKI) we need MultiMap here too though

Reply:
For the thing with LKI trigger, I would need to debug what the right answer should be

Reply:
i think for LKI trigger, only one of them should match? The one with TriggerZone Battlefield?

of the LKI ones should overwrite the other ones